src/CopyrighFixer/CopyrighFixer/owner.cpp

```diff
@@ -20,12 +20,38 @@ const QString &Owner::getOwnerName() const {
     return _name;
 }
 
-void Owner::setTimeRange(const QString &interval) {
-    _timeRange = interval;
+void Owner::setTimePoint(const int &interval) {
+    _timePoint = interval;
 }
 
-const QString &Owner::getTimeRange() const {
-    return _timeRange;
+const int &Owner::getTimePoint() const {
+    return _timePoint;
+}
+
+void Owner::fromjson(const QJsonObject &objJs) {
+    if (objJs.contains("name") && objJs.value("name").isString()) {
+         _name = objJs.value("name").toString();
+    }
+
+    if (objJs.contains("timePoint")) {
+        _timePoint = objJs.value("timePoint").toInt();
+    }
+}
+
+const QJsonObject &Owner::toJson(QJsonObject &objJs) const {
+
+    objJs["name"] = _name;
+    objJs["timePoint"] = _timePoint;
+
+    return objJs;
+}
+
+bool Owner::isValid() const {
+    if (_name.size() != 0 && _timePoint > 0) {
+        return true;
+    }
+
+    return false;
```

Comment:
1. Cool but... why you use the **if** ?
2. And you forget about a check of the time for the future 
```suggestion
    if (_name.size() != 0 && _timePoint > 0) {
        return true;
    }

    return _name.size() && _timePoint > 0 && _timePoint <= time(0);
```